test/functional/rpc_rawtransaction.py

```diff
@@ -178,6 +182,67 @@ def getrawtransaction_tests(self):
         block = self.nodes[0].getblock(self.nodes[0].getblockhash(0))
         assert_raises_rpc_error(-5, "The genesis block coinbase is not considered an ordinary transaction", self.nodes[0].getrawtransaction, block['merkleroot'])
 
+    def getrawtransaction_verbosity_tests(self):
+        tx = self.nodes[2].sendtoaddress(self.nodes[1].getnewaddress(), 1)
+        [block1] = self.generate(self.nodes[2], 1)
+        fields = [
+            'blockhash',
+            'blocktime',
+            'confirmations',
+            'hash',
+            'hex',
+            'in_active_chain',
+            'locktime',
+            'size',
+            'time',
+            'txid',
+            'vin',
+            'vout',
+            'vsize',
+            'weight'
+        ]
+        prevout_fields = [
+            'generated',
+            'height',
+            'value',
+            'scriptPubKey'
+        ]
+        script_pub_key_fields = [
+            'address',
+            'asm',
+            'hex',
+            'type'
+        ]
+        for n in [0, 3]:
+            for v in [1,2]:
+                self.log.info(f"Test getrawtransaction_verbosity {v} {'with' if n == 0 else 'without'} -txindex, with blockhash")
+                gottx = self.nodes[n].getrawtransaction(txid=tx, verbosity=v, blockhash=block1)
+                for f in fields:
+                    if f not in gottx:
+                        raise AssertionError(f"field {f} is not in transaction")
+                    if v == 1:
+                        assert('fee' not in gottx)
+                        assert(len(gottx['vin']) > 0)
```

Comment:
This is not specific to `if v == 1` so I'd move it up one level and remove the duplication from `if v == 2:`